### ISSUE TITLE
Alternative Syntax For Unions in 3.9 triggered unintentionally

### ISSUE THREAD

**Issue by gordonwatts** [microsoft/pyright#689]
**Describe the bug**
`pyright` marks the use of the bit-wise "or" operator, `|`, as a type union operation (to be introduced in 3.9).

**To Reproduce**
Run the below code through `pyright`

**Expected behavior**
Did not expect this particular error report - perhaps another error should have been reported.

**Screenshots or Code**
```
from typing import Any


class my_class:
    def __init__(self, v):
        self._v = v


def test_bad_syntax(a: Any, b: Any):
    r = my_class(a | b)  # pyright finds type error here
    print(r)
```

**VS Code extension or command-line**
VSCode, `pyright` extension version 1.1.39. Python is version 3.7.7

**Additional context**
As always, thanks for this amazing tool!

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.40, which I just published.
